Add "Web Components v0 deprecation" article.

yoichio opened the pull request:
**Target Live Date:** 2018-12-03

- This has NOT been reviewed and approved by anyone.
- [x] I have run `npm test` locally and all tests pass.
- [x] I have added the appropriate `type-something` label.
- [x] I've staged the site and manually verified that my content displays correctly.

**CC:** @petele, @beaufortfrancois, @hayatoito, @tkent-google, @kevinpschaaf, @jpchaseTo

To CCed people: could you review each part of the component that you own?

hayatoito commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+{% include "web/_shared/contributors/yoichiosato.html" %}
+
+A part of web interoperability for [Web Components v1](/web/updates/2017/01/webcomponents-org),
+we're deprecating Shadow DOM v0, Custom Element v0 and HTML Imports in Chrome 73.
```
We're *removing*, instead of *deprecating*, in M73.
They were already deprecated. :)

PaulKinlan replied:
(When roughly will Chrome 73 be available on dev, beta and stable - it would be good to give people an indication of the rough timeline -  generally people don't know the current version we are at so 73 by itself isn't super useful)

yoichio replied:
Done.

hayatoito commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+## Migration support
+If you are using this features, please consider migrating to the new API or
+upgrading your Polymer library.
+Use `--disable-blink-features=ShadowDOMV0,CustomElementsV0,HTMLImports` for
```
It might be better to clarify that this is a command line option to chrome.

yoichio replied:
Oh, I found that devtool didn't work on the chrome. We need to permit devtool accessing the disabled feature.

hayatoito commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+`Element.createShadowRoot()` will be removed. Instead we have `Element.attachShadow()`
+to create Shadow Root.
+If your existing components don't use distribution, replacing with
+`Element.attachShadow({mode:"open"})` might work.
```
Hmm. These are small differences other than distribution, so I don't recommend to say "might work" here. That could be considered as *too optimistic*, I'm afraid.

ebidel replied:
might work -> we recommend...

yoichio replied:
Changed to just say "not backcompatible".

PaulKinlan commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+
+## Reverse Origin Trial
+
+For web authors that don't much time to M73,
```
generally, yes it is fine. I would link to a public timeline to clarify if needed.

beaufortfrancois replied:
We usually refer to Chrome 73 instead of M73.

ebidel replied:
"M" releases are fine, but let's stick with "Chrome 73" for consistency with https://www.chromestatus.com/features/schedule.

PaulKinlan commented:
cc - @ebidel who has a lot of experience with v0

beaufortfrancois commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: A round up of the deprecations and removals in Chrome 71 to help you plan.
```
+1

yoichio replied:
Done.

beaufortfrancois commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+{# wf_featured_image: /web/updates/images/generic/warning.png #}
+{# wf_featured_snippet: Web Components v0 deprecation and removal in Chrome 73 to help you plan.#}
+
+{% include "web/updates/_shared/see-all-dep-rem.html" %}
```
I think* you can safely remove this line as this article is specifically about WebComponents V0.

yoichio replied:
Thanks.
Done.

beaufortfrancois commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+ testing if your site works without those APIs.
+
+## Remove Shadow DOM v0
+`Element.createShadowRoot()` will be removed. Instead we have `Element.attachShadow()`
```
Add extra line.

beaufortfrancois commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+
+
+## Remove Custom Elements v0
+`document.registerElement()` will be replaced by  `window.customElements.define()`.
```
Nit: Add extra line.

ebidel replied:
will be -> is replaced

petele commented:
@ebidel - would you please review and approve, then merge when you're happy.

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+description: A round up of the deprecations and removals in Chrome 71 to help you plan.
+
+{# wf_updated_on: 2018-11-26 #}
+{# wf_published_on: 2018-11-13 #}
```
be sure to update these dates as we get closer to merging.

yoichio replied:
Done.

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+{% include "web/_shared/contributors/yoichiosato.html" %}
+
+A part of web interoperability for [Web Components v1](/web/updates/2017/01/webcomponents-org),
+we're deprecating Shadow DOM v0, Custom Element v0 and HTML Imports in Chrome 73.
```
we're -> Chrome

ebidel replied:
I'd throw in links to the h5r articles for reference:
https://www.html5rocks.com/en/tutorials/webcomponents/shadowdom/
https://www.html5rocks.com/en/tutorials/webcomponents/customelements/
https://www.html5rocks.com/en/tutorials/webcomponents/imports/

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+
+## Migration support
+If you are using this features, please consider migrating to the new API or
+upgrading your Polymer library.
```
no need to call out Polymer explicitly. Just "upgrading your library".

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+
+## Remove HTML Imports
+
+HTML Imports will be removed. Similar functionarity will be served as HTML Modules.
```
functionality

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+
+HTML Imports will be removed. Similar functionarity will be served as HTML Modules.
+Though HTML Module is still under [spec discussion](https://github.com/w3c/webcomponents/issues/645)
+and we don't have any direct replacement for HTML Imports yet, we're deprecating this for healthy web:
```
we're -> Chrome is throughout. Although we (Chrome team) is writing this article, we still like to come across like we're discussing what the "browser vendors" are doing.

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+we're preparing called Reverse [Origin Trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md)
+ where you can use the deprecated features with tokens after M73 for one year
+ to M81.
+Sign up form is coming. Stay tuned.
```
we should really have this ready to go when this article lands.

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,102 @@
+- If you're using Polymer 1.x or 2.x, you'll need to make sure you're loading the polyfills.
+- If you're using Polymer 3.x or LitElement, the deprecation should not affect you.
+
+[Polymer teams' document](https://www.polymer-project.org/blog/2018-10-02-webcomponents-v0-deprecations)
```
how about other libs using v0? We need to not bias towards polymer stuff.

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -1,32 +1,35 @@
 project_path: /web/_project.yaml
 book_path: /web/updates/_book.yaml
-description: A round up of the deprecations and removals in Chrome 71 to help you plan.
+description: Web Components v0 deprecation and removal in Chrome73 to help you plan.
```
Chrome73  -> Chrome 73

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
 [Chromium Bug](https://crbug.com/766694)
 
-### Why at once?
+### Why removing them at once?
```
Why are all APIs being removed at once?

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
- at once, or at least use the polyfill would make it less troublesome, easier to
- migrate, rather than fighting with every edge case of combination of polyfill
-  and native implementation.
+Chrome usually removes features one by one, but why at once for these features this time?
```
but why are these features being removed at the same time?

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
- migrate, rather than fighting with every edge case of combination of polyfill
-  and native implementation.
+Chrome usually removes features one by one, but why at once for these features this time?
+Since launched, we found that they are often used together.
```
we found the v0 APIs are often...

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
-  and native implementation.
+Chrome usually removes features one by one, but why at once for these features this time?
+Since launched, we found that they are often used together.
+Especially [webcomponentsjs polyfill](https://github.com/WebComponents/webcomponentsjs)
```
Especially -> The

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
+Since launched, we found that they are often used together.
+Especially [webcomponentsjs polyfill](https://github.com/WebComponents/webcomponentsjs)
+ was originally designed to simulate the v0 features on other
+browsers and not do so on Chrome being removed each feature randomly.
```
I'm not sure what "and not do so on Chrome being removed each feature randomly" is trying to say.

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
+Especially [webcomponentsjs polyfill](https://github.com/WebComponents/webcomponentsjs)
+ was originally designed to simulate the v0 features on other
+browsers and not do so on Chrome being removed each feature randomly.
+For safer/lesser troublesome migration off from the features, we've desided to
```
For a safer migration

ebidel replied:
For safer/lesser troublesome migration off from the features, we've decided to

note: desided -> decided

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
+ was originally designed to simulate the v0 features on other
+browsers and not do so on Chrome being removed each feature randomly.
+For safer/lesser troublesome migration off from the features, we've desided to
+remove them at one.
```
one -> once

ebidel commented on src/content/en/updates/2018/11/webcomponents-v0-deprecations.md:
```diff
@@ -36,64 +39,62 @@ modules in 2017.
-upgrading your Polymer library.
-Use `--disable-blink-features=ShadowDOMV0,CustomElementsV0,HTMLImports` for
- testing if your site works without those APIs.
+If your site and/or libirary is using this features, you should upgrade them.
```
libirary -> library.

Can you give the article a proofread for typos. I'll hold off on reviewing more.

petele commented:
@yoichio - any updates on this?

tkent-google commented:
@yoichio is on vacation. Probably no update until the next year.

yoichio commented:
Moved the article to 2019/01 so most of comment will be outdated though I addressed them.
Preparing Blink code for M73 removal is not yet finished but I want this article ready so let's go on review.

clelland commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,103 @@
+
+### Reverse Origin Trial
+
+For web authors that don't much time to Mar, 2019 at Chrome 73,
```
or maybe "for up to an additional year", if the tokens aren't going to last for the entire year. Do we know if that has been decided yet?

yoichio replied:
Shorter is better, of course. Chrome M81 is planned to be the first Chrome that has no capability to access the APIs .

yoichio commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,103 @@
+### Why are all APIs being removed at once?
+
+Chrome usually removes features one by one, but why are these features being removed at the same time?
+Since launched, we found that the v0 APIsy are often used together.
```
Done. I understand the circumstance that we don't want to say "because of poly*" no matter that's good or bad.

yoichio commented:
Update.
I guess I should update my patch w/o rebase -i for review tracking,,,(I'm not familiar with github workflow)?

yoichio commented:
Updated.
I guess I should update my patch w/o rebase -i for review tracking,,,(I'm not familiar with github workflow)?

jpchase commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,99 @@
+[AshleyScirra’s HTML Imports polyfill](https://github.com/AshleyScirra/html-imports-polyfill)).
+
+
+### Reverse Origin Trial
```
```suggestion
### Postpone removal via origin trial
```
Suggesting a different heading:
- I'd prefer not to introduce a term like "reverse origin trial" that could be difficult for developers to understand.
- Minor nit is that "origin trials" is generally not capitalized. It is not a branded name.

jpchase commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,99 @@
+### Reverse Origin Trial
+
+For web authors who need more time to handle this removal, you can re-enable
+these features with an [Origin Trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md).
```
```suggestion
these features with an [origin trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md).
```
Lower case for "origin trial"

jpchase commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,99 @@
+
+For web authors who need more time to handle this removal, you can re-enable
+these features with an [Origin Trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md).
+With a token generated for the trial, your site can continue to use them for
```
```suggestion
After registering for the trial, you will receive a token that allows your site to use the three deprecated features for up to a year,
```

jpchase commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,99 @@
+For web authors who need more time to handle this removal, you can re-enable
+these features with an [Origin Trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md).
+With a token generated for the trial, your site can continue to use them for
+an additional year, until Chrome 81.
```
```suggestion
until Chrome 81.
```

jpchase commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,99 @@
+
+The trial name is `WebComponentsV0`. With that token, you can use Shadow DOM v0,
+Custom Elements v0, and HTML Imports APIs. Trial registration will be open
+after Chrome 73 beta release."
```
I suggest simplifying the preceding two paragraphs to something like:

```suggestion
We plan to remove the features permanently in the Chrome 81 release, although the trial tokens may expire before then.

Soon after the Chrome 73 beta release, you can register for the trial at [Web Components V0](https://developers.chrome.com/origintrials/#/view_trial/2431943798780067841).
```
- The text above (with suggested edits) already explains that the trial will re-enable all 3 features, so probably don't need to repeat that here.
- I've added the link to the specific trial. This link will not work until the trial is made public, but it will redirect to the list of trials in the meantime.

jpchase commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,101 @@
+the trial token might last earlier.
+
+The trial name is `WebComponentsV0`. With that token, you can use Shadow DOM v0,
+Custom Elements v0, and HTML Imports APIs. Trial registration will be open
```
My earlier [comment](https://github.com/google/WebFundamentals/pull/6890#discussion_r247648880) could have been more clear. I suggested new content, but that intended to replace existing paragraphs, not as an addition. I see the suggested content was added, so now need to remove the two paragraphs, starting with "Chrome 81 is the .." and ending with "... registration will be open" (lines 65 - 69 in the current version of the file).

yoichio replied:
Done

yoichio commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,101 @@
+these features with an [origin trial](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md).
+After registering for the trial, you will receive a token that allows your site to use the three deprecated features for up to a year,
+until Chrome 81.
+Chrome 81 is the planned milestone when we remove implementation code entirely so
```
```suggestion
```

yoichio commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,101 @@
+After registering for the trial, you will receive a token that allows your site to use the three deprecated features for up to a year,
+until Chrome 81.
+Chrome 81 is the planned milestone when we remove implementation code entirely so
+the trial token might last earlier.
```
```suggestion
```

yoichio commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,101 @@
+until Chrome 81.
+Chrome 81 is the planned milestone when we remove implementation code entirely so
+the trial token might last earlier.
+
```
```suggestion
```

yoichio commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,101 @@
+Chrome 81 is the planned milestone when we remove implementation code entirely so
+the trial token might last earlier.
+
+The trial name is `WebComponentsV0`. With that token, you can use Shadow DOM v0,
```
```suggestion
```

yoichio commented on src/content/en/updates/2019/01/webcomponents-v0-deprecations.md:
```diff
@@ -0,0 +1,101 @@
+the trial token might last earlier.
+
+The trial name is `WebComponentsV0`. With that token, you can use Shadow DOM v0,
+Custom Elements v0, and HTML Imports APIs. Trial registration will be open
```
```suggestion
```

yoichio commented:
Thanks everyone for reviewing this and preparation work.
Since It turned out that many Chrome internal components using the APIs didn't yet move away from them,
I decided to postpone API removal to M74.

yoichio commented:
The removing patch was landed:
https://bugs.chromium.org/p/chromium/issues/detail?id=911943#c21

I also updated date/milestone of my article.
Could you review to merge?

yoichio commented:
Removal was postponed. So plz wait to merge.

petele commented:
Any update on this? Should we be keeping it?

yoichio commented:
Since an alternative article for this topic was published: https://developers.google.com/web/updates/2019/07/web-components-time-to-upgrade
let me close this. Thanks.
